### DISCUSSION TITLE
how can i add this parameters to chat-persistent.sh

### DISCUSSION THREAD

**Discussion by kroryan** [ggml-org/llama.cpp#2862] [General]
Hello, im trying to add the parameters -n 256 and --keep 48 and --color -i on this [https://github.com/kroryan/llama/blob/main/chat-persistent.sh](chat-persistent.sh) somebody can help me? i dont know where i have to write it and im getting errors

**Comment by KerfuffleV2** [comment]
Don't think you can. It has its own logic for context swapping which will probably conflict with trying to change stuff like `-n` or `--keep`. Not sure about interactive mode, but it does stuff like parse the log and it may be different if you turn on interactive. Same for `--color`: that could mess up the parsing it does to check for stuff.

This is where it runs `main`: https://github.com/ggerganov/llama.cpp/blob/bcce96ba4dd95482824700c4ce2455fe8c49055a/examples/chat-persistent.sh#L112-L117

I don't recommend messing with it and changes are very likely to cause weird issues. If you want to though, you can try adding something like `-i` right before one of the other options like `--n_predict`
